middle_end/flambda/to_cmm/un_cps_helper.ml

```diff
@@ -143,7 +143,11 @@ let and_ = binary Cmm.Cand
 let or_ = binary Cmm.Cor
 let xor_ = binary Cmm.Cxor
 
-let eq = binary Cmm.(Ccmpi Ceq)
+let eq ?(dbg=Debuginfo.none) x y =
+  match x, y with
+  | Cmm.Cconst_int (n, _), Cmm.Cop(Csubi, [Cconst_int (m, _); c], _) ->
```

Comment:
Two remarks:
- Could add a comment here explaining the kind of pattern that is optimized here (i.e. the source boolean negation, ultimately translated as a `(= 1 (4 - c))` )
- Could you also add the symmetric case ? Nothing really guarantees the current ordering of comparison and it seems a very likely pitfall to only do the optimization when arguments are in this order.